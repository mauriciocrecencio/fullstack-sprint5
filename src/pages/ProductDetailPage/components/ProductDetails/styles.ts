import styled from 'styled-components'

export const Container = styled.div`
  width: 45%;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  position: relative;
`
export const ProductName = styled.span`
  margin-top: -50px;
  font-size: 1.6875rem;
  line-height: 1.167;
  color: #261e1e;
`
