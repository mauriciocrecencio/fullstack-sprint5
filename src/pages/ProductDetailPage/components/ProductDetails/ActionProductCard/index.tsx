import { useHistory } from 'react-router-dom'
import * as S from './styles'

interface IPropsActionProductCard {
  price: string
}

const ActionProductCard = ({ price }: IPropsActionProductCard) => {
  const history = useHistory()
  return (
    <S.Container>
      <S.ProductPrice>R$ {price}</S.ProductPrice>
      <S.AddCartButton onClick={() => history.push('/')}>
        Adicionar Ã  sacola
      </S.AddCartButton>
      <S.CancelButton onClick={() => history.push('/')}>
        Cancelar
      </S.CancelButton>
    </S.Container>
  )
}

export default ActionProductCard
