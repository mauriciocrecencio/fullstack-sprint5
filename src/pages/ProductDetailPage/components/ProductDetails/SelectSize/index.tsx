import * as S from './styles'

interface IPropsSizeBox {
  size: number
}

const sizes = [6, 7, 8, 10, 12, 14, 15]
const SizeBox = ({ size }: IPropsSizeBox) => {
  return <S.SizeBoxContainer>{size}</S.SizeBoxContainer>
}

const SelectSize = () => {
  return (
    <>
      <S.SelectSizeText>Selecionar Tamanho: 6</S.SelectSizeText>
      <S.Container>
        {sizes.map((size,index) => (
          <SizeBox key={index} size={size} />
        ))}
      </S.Container>
    </>
  )
}

export default SelectSize
