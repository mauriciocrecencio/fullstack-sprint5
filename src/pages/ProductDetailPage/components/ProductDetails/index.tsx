import * as S from './styles'
import ActionProductCard from './ActionProductCard'
import SelectSize from './SelectSize'

interface IPropsProductDetails {
  name: string
  price: string
}

const ProductDetails = ({ name, price }: IPropsProductDetails) => {
  return (
    <S.Container>
      <S.ProductName>{name}</S.ProductName>
      <SelectSize />
      <ActionProductCard price={price} />
    </S.Container>
  )
}

export default ProductDetails
