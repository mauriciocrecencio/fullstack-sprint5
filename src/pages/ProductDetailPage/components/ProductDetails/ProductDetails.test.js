/* eslint-disable no-undef */
import { render } from '@testing-library/react'
import ProductDetails from './index'

describe('Componente pÃ¡gina do produto', () => {
  it('O snapshot do component deve permanecer sempre o mesmo', () => {
    const { container } = render(
      <ProductDetails name="Nome teste" price="R$3,40" />
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})

export {}
