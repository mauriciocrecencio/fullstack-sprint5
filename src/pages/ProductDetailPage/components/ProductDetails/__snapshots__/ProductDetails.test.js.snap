// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componente página do produto O snapshot do component deve permanecer sempre o mesmo 1`] = `
<div
  class="sc-bdnxRM jpmkCl"
>
  <span
    class="sc-gtsrHT cbMPhD"
  >
    Nome teste
  </span>
  <span
    class="sc-jrsJWt glsTHx"
  >
    Selecionar Tamanho: 6
  </span>
  <div
    class="sc-pNWdM jrasAX"
  >
    <div
      class="sc-kEqXSa dOoelZ"
    >
      6
    </div>
    <div
      class="sc-kEqXSa dOoelZ"
    >
      7
    </div>
    <div
      class="sc-kEqXSa dOoelZ"
    >
      8
    </div>
    <div
      class="sc-kEqXSa dOoelZ"
    >
      10
    </div>
    <div
      class="sc-kEqXSa dOoelZ"
    >
      12
    </div>
    <div
      class="sc-kEqXSa dOoelZ"
    >
      14
    </div>
    <div
      class="sc-kEqXSa dOoelZ"
    >
      15
    </div>
  </div>
  <div
    class="sc-jSFjdj kFFoyY"
  >
    <span
      class="sc-gKAaRy iwCTiL"
    >
      R$ 
      R$3,40
    </span>
    <button
      class="sc-eCApnc sc-iCoGMd cbWHgP gfXkgX"
    >
      Adicionar à sacola
    </button>
    <button
      class="sc-eCApnc sc-fujyAs cbWHgP hxVucD"
    >
      Cancelar
    </button>
  </div>
</div>
`;
